scrapy_azure_exporter/azure_exporter.py

```diff
@@ -1,45 +1,113 @@
-from azure.storage.blob import BlobServiceClient
-from azure.core.exceptions import ResourceNotFoundError
+import logging
+import os
 import typing
+from urllib.parse import urlparse
+
+from azure.core.exceptions import AzureError
+from azure.storage.blob import BlobServiceClient
+from azure.storage.blob._models import BlobType
 from scrapy.exceptions import NotConfigured
 from scrapy.extensions.feedexport import BlockingFeedStorage
-import logging
 
-"""The pipeline for adding a file to Azure Blob"""
 
+class AzureFeedStorage(BlockingFeedStorage):
+    """This class will be used for exporting files to Azure Blob Storage"""
 
-class AzureFeedExporter(BlockingFeedStorage):
+    def __init__(
+        self,
+        uri,
+        connection_string,
+        account_url_with_sas_token,
+        account_url,
+        account_key,
+        *,
+        feed_options=None,
+    ):
+        if feed_options is None:
+            feed_options = {}
+        self.connection_string = connection_string
+        self.account_url_with_sas_token = account_url_with_sas_token
+        self.account_url = account_url
+        self.account_key = account_key
+        self.overwrite = feed_options.get("overwrite")
+        self.blob_type = feed_options.get("blob_type", "BlockBlob")
 
-    def __init__(self, container_name, blob_service_client, azure_export_filename):
```

Comment:
Shall we make this change backward-compatible, make it so that old method parameters (and Scrapy settings) still work as before, but trigger a deprecation warning?

Reply:
Hi @Gallaecio, that's a significant consideration.
However, previously the exporter wasn't working and I couldn't make it work in that state. So making this change backward-compatible seems to be not so useful. Feel free to suggest if you have any ideas. 
Thanks.